common/types/encoding/codecv1/codecv1.go

```diff
@@ -0,0 +1,450 @@
+package codecv1
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+	"strings"
+
+	"github.com/scroll-tech/go-ethereum/accounts/abi"
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/core/types"
+	"github.com/scroll-tech/go-ethereum/crypto"
+	"github.com/scroll-tech/go-ethereum/crypto/kzg4844"
+	"github.com/scroll-tech/go-ethereum/log"
+
+	"scroll-tech/common/types/encoding"
+)
+
+var (
+	// BLSModulus is the BLS modulus defined in EIP-4844.
+	BLSModulus *big.Int
+
+	// BlobDataProofArgs defines the argument types for `_blobDataProof` in `finalizeBatchWithProof4844`.
+	BlobDataProofArgs abi.Arguments
+
+	// MaxNumChunks is the maximum number of chunks that a batch can contain.
+	MaxNumChunks int = 15
```

Comment:
Would this value change in this codec version?

Reply:
I don't think so. If we change it, we'll also need to change the provers, which will require another codec version.

Reply:
Got it.